src/gan/gen_pyt/model/parser.py

```diff
@@ -213,27 +216,37 @@ def encode(self, src_sents_var, src_sents_len):
             src_sents_var = src_sents_var * mask + (1 - mask) * self.vocab.unk
 
         src_token_embed = self.src_embed(src_sents_var)
-        packed_src_token_embed = pack_padded_sequence(src_token_embed, 
-                                                      src_sents_len)
+
+        src_encodings = self.encoder_cnn(src_token_embed)
+
+        src_encodings = src_encodings.permute(1, 0, 2)
+
+        # packed_src_token_embed = pack_padded_sequence(src_token_embed, 
+        #                                               src_sents_len)
 
         # src_encodings: (tgt_query_len, batch_size, hidden_size)
-        src_encodings, (last_state, last_cell) = self.encoder_lstm(packed_src_token_embed)
-        src_encodings, _ = pad_packed_sequence(src_encodings)
+        last_cell = src_encodings[:, -1, :]
+        last_state = src_encodings[: , -1, :]
+
```

Comment:
So, this makes intuitive sense to me, but when I run the current LSTM-encoded version this doesn't seem to be what's really happening. `last_cell` does not end up matching the last slice from `src_encodings` - I think because it's a bi-directional LSTM? 

I think we try a `MaxPool1d` layer here, sort of a single-value decomposition of the sequences. Something like the below would give us the right shape coming out (`(batch_size, hidden_size)`):
```python
last_cell = nn.MaxPool1d(src_encodings.shape[1])(src_encodings)
last_cell = last_cell.reshape(self.args['batch_size'], self.args['hidden_size'])
```
(The argument to `MaxPool1d` is the kernel size to do the calculation over, and if we want one value for the sequence we give it the length of the sequence, which is the second dimension of `src_encodings`. We then reshape that to be `(batch_size, hidden_size)` to match the expected shape going forward.)

Reply:
One note: sorry, this would assume we do that `src_encodings.permute(0, 2, 1)` after the `MaxPool1d` - I'm basing this off of the comments on tensor shapes in the code as is.

Reply:
Hi Ben, 

I reviewed this again and chose to use: last_cell = torch.max(src_encodings, 1).
MaxPool1d based on (10*14*256) didn't provide us (10*256), but I used torch.max to change (10*14*256) to (10*256) -basically take the max value -- this might be done by MaxPool but I got confused. I thought this approach solved the issue, and proceeded with pretraining successfully, but let me know if you think there is an issue with this part so we can discuss.